### PR TITLE
Add unsigned to signed tests for opening hours

### PR DESCRIPTION
**Pull request by peternewman**
As I suggested in https://github.com/streetcomplete/StreetComplete/pull/4276#pullrequestreview-1064477632

### CODE REVIEW

**Comment by peternewman** [comment]
Test run here: https://github.com/peternewman/StreetComplete/runs/7816823231?check_suite_focus=true

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 1]
```diff
@@ -47,6 +48,40 @@ class AddOpeningHoursTest {
...
+
+    @Test fun `apply same description answer again when it was not signed before`() {
+        questType.verifyAnswer(
+            mapOf(
+                "opening_hours" to "\"oh\"",
+                "opening_hours:signed" to "no"
+            ),
+            DescribeOpeningHours("oh"),
+            StringMapEntryModify("opening_hours", "\"oh\"", "\"oh\""),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString()),
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:80]: It feels to me like technically it shouldn't tag the check_date here given other fields have changed (given that's how SC operates in general now), but I appreciated that's all a bit complicated when it's a related field that's changed.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 2]
```diff
@@ -64,6 +99,15 @@ class AddOpeningHoursTest {
         )
     }
 
+    @Test fun `apply no opening hours sign answer when there was an always open answer before`() {
+        questType.verifyAnswer(
+            mapOf("opening_hours" to "24/7"),
+            NoOpeningHoursSign,
+            StringMapEntryAdd("opening_hours:signed", "no"),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString())
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:107]: Likewise, and more so here.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 3]
```diff
@@ -88,6 +132,40 @@ class AddOpeningHoursTest {
...
+
+    @Test fun `apply always open answer when it was not signed but present before`() {
+        questType.verifyAnswer(
+            mapOf(
+                "opening_hours" to "24/7",
+                "opening_hours:signed" to "no"
+            ),
+            AlwaysOpen,
+            StringMapEntryModify("opening_hours", "24/7", "24/7"),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString()),
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:152]: Plus here, again we've only checked the unsigned aspect.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 4]
```diff
@@ -151,6 +229,79 @@ class AddOpeningHoursTest {
...
+                        it.startDay = WeekDay.MO
+                    })
+                    times = listOf(TimeSpan().also {
+                        it.start = 60 * 10
+                        it.end = 60 * 12
+                    })
+                })
+            )),
+            StringMapEntryModify("opening_hours", "Mo 10:00-12:00", "Mo 10:00-12:00"),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString()),
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:300]: Exactly the same as the one above.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 5]
```diff
@@ -64,6 +99,15 @@ class AddOpeningHoursTest {
...
 
+    @Test fun `apply no opening hours sign answer when there was an always open answer before`() {
+        questType.verifyAnswer(
+            mapOf("opening_hours" to "24/7"),
+            NoOpeningHoursSign,
+            StringMapEntryAdd("opening_hours:signed", "no"),
+            StringMapEntryAdd("check_date:opening_hours", LocalDate.now().toCheckDateString())
+        )
+    }
+
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:110]: I've just realised that all apart from this one are ineligible for this quest as they'll be filtered out for being unsigned already.

I should be able to apply most to https://github.com/streetcomplete/StreetComplete/blob/master/app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours_signed/CheckOpeningHoursSignedTest.kt though.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 1 above)
Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:80]: yeah

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 3 above)
Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:152]: What? No. here we confirmed that it is open 24/7

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 5 above)
Comment (westnordost) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:110]: Delete them then? Or move them to CheckOpeningHoursSignedTest?

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 3 above)
Comment (peternewman) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:152]: Oops yeah.

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 5 above)
Comment (peternewman) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:76]: I've migrated some and reworked others and deleted some others to make them all relevant and appropriate to the various quests.

As part of #4276 they could be extended further/rewritten.

**Comment by peternewman** [comment]
Test run: https://github.com/peternewman/StreetComplete/runs/7824199849?check_suite_focus=true

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt (hunk 2 above)
Comment (matkoniecz) [reply app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:73]: maybe mention it is some comment why SC is doing this? Or point to the comment explaining what is going on?

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours_signed/CheckOpeningHoursSignedTest.kt [hunk 6]
```diff
@@ -85,4 +133,29 @@ class CheckOpeningHoursSignedTest {
...
+    @Test fun `apply no answer with prior check date and existing opening hours via other means`() {
+        questType.verifyAnswer(
+            mapOf(
+                "opening_hours" to "Mo 10:00-12:00",
+                "opening_hours:signed" to "no",
+                "check_date:opening_hours" to "2020-03-04"
+            ),
+            false,
+            StringMapEntryModify("opening_hours:signed", "no", "no"),
+            StringMapEntryModify("check_date:opening_hours", "2020-03-04", LocalDate.now().toCheckDateString()),
```
Comment (matkoniecz) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours_signed/CheckOpeningHoursSignedTest.kt:158]: this also seems weird, and ideally would be mentioned why it is not using `check_date:opening_hours:signed` (but can be treated as out of scope of that PR)

**Review by westnordost — APPROVED** [review]
Nice, easy to read.

The tests you wrote document current behavior, so (to discuss/solve) any weirdness in the current behavior is out of scope of this PR. After all, there is already a ticket for this: #4273

File: app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt [hunk 7]
```diff
@@ -5,6 +5,7 @@ import ch.poole.openinghoursparser.TimeSpan
 import ch.poole.openinghoursparser.WeekDay
 import ch.poole.openinghoursparser.WeekDayRange
 import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapEntryAdd
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapEntryDelete
```
Comment (peternewman) [inline app/src/test/java/de/westnordost/streetcomplete/quests/opening_hours/AddOpeningHoursTest.kt:8]: I've just realised this import is now redundant after the changes. 🤦

**Pull request merged**